src/js/components/sidebar.jsx

```diff
@@ -97,37 +100,58 @@ export default class SideBar extends React.Component {
       filter,
       filteredTabs,
       selectAll,
+      tabIds,
       tabsById,
       moreActionsShown,
     } = this.state;
 
     const tabs = filteredTabs.map(tabId => {
       const tabInfo = tabsById.get(tabId);
+      const tabFilteredIndex = filteredTabs.indexOf(tabId);
+
+
       return (
         <TabInfo
           key={tabInfo.id}
           active={activeTabId === tabId}
           tabInfo={tabInfo}
           onClick={this._onTabInfoClicked}
           onSelectionChanged={this._onSelectionChanged}
+          pinned={tabInfo.pinned}
         />
       );
     });
 
+    const pinnedInfoTitle = this._hasPinnedTabs(filteredTabs)
+      ? <li id="pinInfo">Pinned tabs</li>
+      : null;
+
     const selectedTabsCount = this._getSelectedTabIds().length;
     const moreActionsClasses = !moreActionsShown ? "hidden" : "";
 
+    const hasSelectedPinnedTabs = this._hasPinnedTabs(
+      this._getSelectedTabIds(),
+    );
+
     return (
       <div id="sidebar">
         <div id="controls">
           <div id="counter">
             {selectedTabsCount} / {this.state.tabIds.length} tabs selected
           </div>
           <div id="actions">
-            <button id="close" onClick={() => this._closeSelected()}>
+            <button
+              id="close"
+              onClick={() => this._closeSelected()}
+              disabled={hasSelectedPinnedTabs}
+            >
```

Comment:
Nit: I'd kinda prefer to put the closing < on the same line as the last attribute. I know this adds to blame if we add a new last attribute... but we can always add attributes to the middle of the list. Same below.